core/lib/Cardano/Address/Script.hs

```diff
@@ -306,6 +380,10 @@ data ErrValidateScript
     | DuplicateSignatures
     | WrongKeyHash
     | Malformed
+    | InvalidTimelocks
+    | DuplicateXPubs
```

Comment:
Couldn't this case be solved by modelling the ScriptTemplate differently? For instance, use a `Set (Cosigner, XPub)`. In practice, the set will be reasonably bounded (and we could/should very much enforce that) so searching in the set for a particular index would be fast. Plus, it automatically enforces uniqueness of elements.

Reply:
yes, I used set operation to resolve this. please refer to HEAD